x-pack/plugins/search_notebooks/public/plugin.ts

```diff
@@ -13,13 +13,16 @@ import {
   SearchNotebooksPluginSetup,
   SearchNotebooksPluginStart,
   SearchNotebooksPluginStartDependencies,
+  NotebookListValue,
 } from './types';
 import { getErrorCode, getErrorMessage, isKibanaServerError } from './utils/get_error_message';
 
 export class SearchNotebooksPlugin
   implements Plugin<SearchNotebooksPluginSetup, SearchNotebooksPluginStart>
 {
+  private notebooksList: NotebookListValue = null;
```

Comment:
I tried to use an Observable for this, but found it lacking. the `useObservable` hook initializes to `undefined` so you then have to have logic to wait for a re-render to get the actual value. This was not ideal for the hook that fetches the catalog. So I opted for storing the state in a variable and passing around the get function instead.